src/server/fileio.js

```diff
@@ -35,9 +37,59 @@ export const configurableUploadHandler = (req, res, apiBaseURL) => {
         })
     ).pipe(res);
 };
+
 /**
  * Same as configurableUploadHandler, but defaults to using the configured
  * terrain handler.
  */
 export const uploadHandler = (req, res) =>
     configurableUploadHandler(req, res, terrainURL);
+
+/**
+ * Handles proxying file download requests to the Terrain API.
+ * @param {object} req - An Express Request object.
+ * @param {object} res - An Express Response object.
+ * @param {string} apiBaseURL - The base URL to the backing API. Usually an instance of the terrain API.
+ * @returns null
+ */
+export const configurableDownloadHandler = (req, res, apiBaseURL) => {
+    const token = req?.user?.accessToken;
+
+    if (!token) {
+        res.status(401).send("Authorization required.");
+        return;
+    }
+
+    const { source } = req.query;
+    if (!source || source === "") {
+        res.status(400).send("source query parameter must be set.");
+        return;
+    }
+
+    const apiURL = new URL(apiBaseURL);
+    apiURL.pathname = path.join(apiURL.pathname, "/secured/fileio/download");
+    apiURL.search = `path=${source}`;
+
+    console.log(apiURL.href);
```

Comment:
Someone (I think Ian) mentioned that he liked the fact that backend request URLs were logged by Apollo, so I added the logging here since this code bypasses Apollo.